test/unit/lodash.spec.js

```diff
@@ -0,0 +1,605 @@
+    //         params = datasets.setWith();
+    //     });
+    //     it('should match types of passing elements', function() {
+    //         var element1 = params[0];
+    //         var element2 = params[1];
+    //         var element4 = params[3];
+    //         expect(element1).to.be.an('object');
+    //         expect(element2).to.be.a('string');
+    //         expect(element4).to.be.a('function');
+    //         expect(params.length).to.eql(4);
+    //     });
+    //     it('should set height value of rect1 to square of 7', function() {
+    //         expect(_.setWith.apply(_, params)).to.eql({
+    //             'number': 49
+    //         });
+    //     });
+    // });
+
+    describe('toPairs', function() {
+        beforeEach(function() {
+            params = datasets.toPairs();
+        });
+        it('should be an object', function() {
+            var e1 = params[0];
+            expect(e1).to.be.an('object');
+        });
+        it('should contain property but not own property', function() {
+            var obj = params[0];
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    expect(key).to.not.eql('red');
+                }
+            }
+            expect(obj).to.have.property('cherry');
+        });
+        it('should contain only 2 own properties', function() {
+            var obj = params[0];
+            expect(Object.keys(obj).length).to.eql(2);
+        });
+        it('should return new array of key-value pairs', function() {
+            expect(_.toPairs.apply(_, params)).to.eql([
+                ['apple', 'green'],
+                ['banana', 'yellow']
+            ]);
+        });
+    });
+
+    describe('toPairsIn', function() {
+        beforeEach(function() {
+            params = datasets.toPairsIn();
+        });
+        it('should be an object', function() {
+            var e1 = params[0] instanceof Object;
+            expect(e1).to.eql(true);
+        });
+        it('should contain 2 properties', function() {
+            var obj = params[0];
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    expect(key).to.not.eql('red');
+                }
+            }
+            expect(Object.keys(obj).length).to.eql(1);
+        });
+        it('should return new array of key-value pairs', function() {
+            expect(_.toPairs.apply(_, params)).to.eql([
+                ['weight', 0.2]
+            ]);
+        });
+    });
+
+    describe('transform', function() {
+
+        describe('transform1', function() {
+            beforeEach(function() {
+                params = datasets.transform1();
+            });
+            it('should match types of passing elements', function() {
+                var element1 = params[0];
+                var element2 = params[1];
+                expect(element1).to.be.an('array');
+                expect(element2).to.be.a('function');
```

Comment:
Why do we need to check that if API clearly states that function is the only possible argument??